src/script/search/SearchRepository.ts

```diff
@@ -99,13 +105,20 @@ export class SearchRepository {
       return emojis;
     }, {});
     const termSlug = computeTransliteration(term, excludedEmojis);
-    const weightedResults = userEntities.reduce((results, userEntity) => {
-      const values = properties
+    const weightedResults = userEntities.reduce<{user: User; weight: number}[]>((results, userEntity) => {
+      /*
+        given user of name Bardia and username of bardia_wire this mapping
+        will get name & username properties and return an array value like ['Bardia', 'bardia_wire']
+      */
+      const values: string[] = properties
         .slice()
         .reverse()
-        .map((property: keyof User) =>
-          typeof userEntity[property] === 'function' ? (userEntity[property] as Function)() : userEntity[property],
-        );
+        .map(prop => {
+          const property = prop as keyof User;
+          return typeof userEntity[property] === 'function'
+            ? (userEntity[property] as Function)()
+            : userEntity[property];
```

Comment:
curious about why this section changed. id assume including an extra 'as' would be slightly less precisely typed but maybe there was a reason?

Reply:
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/63250054/217185820-262ce9a3-bd16-4d19-8313-2b9b0c4a153c.png">
we have this error because ts-compiler is unable to recognize property is a key of User, had to change it to this to fix the error.